rust_src/src/lisp.rs

```diff
@@ -106,11 +142,33 @@ impl<T> ExternalPtr<T> {
     pub const fn new(p: *mut T) -> Self {
         Self(p)
     }
+
+    pub fn is_null(self) -> bool {
+        self.0.is_null()
+    }
+
+    pub const fn as_ptr(self) -> *const T {
+        self.0
+    }
+
+    pub fn as_mut(&mut self) -> *mut T {
+        self.0
+    }
+
+    pub fn from_ptr(ptr: *mut c_void) -> Option<Self> {
+        unsafe { ptr.as_ref().map(|p| mem::transmute(p)) }
+    }
+}
+
+impl<T> Deref for ExternalPtr<T> {
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*self.0 }
+    }
 }
 
-pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;
-impl LispSubrRef {
-    pub fn as_mut(self) -> *mut Aligned_Lisp_Subr {
-	self.0
```

Comment:
@DavidDeSimone I added it to `ExternalPtr<T>`

Reply:
That looks good